dragonchain/lib/logs.py

```diff
@@ -0,0 +1,26 @@
+import os
+import json
+from typing import List, Dict, Optional, cast, Any
+
+import requests
+
+from dragonchain.lib.faas import get_faas_auth
+
+FAAS_GATEWAY = os.environ["FAAS_GATEWAY"]
+
+
+def get_raw_logs(contract_id: str, since: Optional[str] = None, tail: Optional[int] = 100) -> List[str]:
+    """Calls openfaas /system/logs endpoint with query parameters for a specific contract"""
+    endpoint = f"{FAAS_GATEWAY}/system/logs"
+    query_params = cast(Dict[str, Any], {"name": f"contract-{contract_id}", "tail": tail, "since": since})
+    response = requests.get(endpoint, params=query_params, headers={"Authorization": get_faas_auth()})
+    if response.status_code != 200:
+        raise RuntimeError("Error getting contract logs, non-2XX response from OpenFaaS gateway")
```

Comment:
We should probably consider having a faas error or something so that's it's easily catchable if we try to catch in the future